src/key_builder.rs

```diff
@@ -328,49 +354,57 @@ impl KeyBuilder {
         self.keypath.push(escaped_key);
     }
 
+    /// adds array to keypath
```

Comment:
Minor nitpick. I always start comments uppercase and most are. If you don't feel like it, you don't have to change it, it's just an info for future comments.